### PR TITLE
remove linked proxy mode

### PR DESCRIPTION
**Pull request by boyamurthy**
remove linked proxy mode for PowerMax

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/757 |

# Checklist:

- [X] Have you run format,vet & lint checks against your submission?
- [X] Have you made sure that the code compiles?
- [X] Did you run the unit & integration tests successfully?
- [ ] Have you maintained at least 90% code coverage?
- [ ] Have you commented your code, particularly in hard-to-understand areas
- [ ] Have you done corresponding changes to the documentation
- [X] Did you run tests in a real Kubernetes cluster?
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

### CODE REVIEW

File: csireverseproxy/main.go [hunk 1]
```diff
@@ -278,20 +267,7 @@ func (s *Server) EventHandler(k8sUtils k8sutils.UtilsInterface, secret *corev1.S
...
-				log.Errorf("failed to get cert file from secret (error: %s). ignoring the config change event", err.Error())
-				return
-			}
-			isUpdated := conf.LinkProxyConfig.UpdateCertFileName(secret.Name, certFileName)
-			if isUpdated {
-				hasChanged = true
-			}
-		}
-	} else {
+	if conf.Mode == config.StandAlone {
```
[the remark this answers is not in the document]
Comment (delldubey) [reply csireverseproxy/main.go:270]: We can have the check till we have a mode in values.

**Review by delldubey — APPROVED** [review]
lgtm

File: csireverseproxy/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (boyamurthy) [reply csireverseproxy/main.go]: fixed the comments .

File: csireverseproxy/main_test.go [hunk 2]
```diff
@@ -61,12 +60,11 @@ func (mock *mockServer) certPEMBytes() []byte {
 }
 
 const (
-	tmpLinkedConfigFile    = "linked-config-test-main.yaml"
-	tmpSAConfigFile        = "sa-config-test-main.yaml"
-	authenticationEndpoint = "/univmax/authenticate"
-	timeoutEndpoint        = "/univmax/timeout"
-	defaultEndpoint        = "/univmax"
-	// Linked proxy config constants
+	tmpLinkedConfigFile       = "linked-config-test-main.yaml"
```
[the remark this answers is not in the document]
Comment (boyamurthy) [reply csireverseproxy/main_test.go]: yes , it is required .

**Review by delldubey — APPROVED** [review]
lgtm

**Review by delldubey — APPROVED** [review]
lgtm

**Review by tssushma — APPROVED** [review]
lgtm

File: csireverseproxy/deploy/config.yaml [hunk 3]
```diff
@@ -1,19 +1,7 @@
-mode: Linked
+mode: StandAlone
```
Comment (shanmydell) [inline csireverseproxy/deploy/config.yaml:1]: @boyamurthy : do we need the mode parameter itself?

**Review by shanmydell — CHANGES_REQUESTED** [review]

File: csireverseproxy/deploy/config.yaml (hunk 3 above)
Comment (delldubey) [reply csireverseproxy/deploy/config.yaml:1]: Yes, this is needed.

File: csireverseproxy/main_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (delldubey) [reply csireverseproxy/main_test.go]: This is removed now.

**Review by shanmydell — APPROVED** [review]

**Pull request merged**